pkg/compactor/block_upload.go

```diff
@@ -36,6 +40,22 @@ const (
 	validationFilename    = "validation.json"
 )
 
+type BlockUploadConfig struct {
+	ValidationType              int           `yaml:"-"`
+	ValidationHeartbeatInterval time.Duration `yaml:"validation_heartbeat_interval" category:"experimental"`
+	ValidationHeartbeatTimeout  time.Duration `yaml:"validation_heartbeat_timeout" category:"experimental"`
+}
+
```

Comment:
I don't think we need to make these configurable for the user. Reasonable default values are fine.